src/java.logging/share/classes/java/util/logging/StreamHandler.java

```diff
@@ -184,10 +184,14 @@ public synchronized void setEncoding(String encoding)
      *                 silently ignored and is not published
```

Comment:
There needs to be an `@implSpec` somewhere in this method's javadoc comment that explains the locking policy here very crisply for subclassers. Specifically (1) the lock on `this` is not held during formatting; (2) the lock on `this` is held during publishing; (3) subclassers must not lock on `this` while calling super.publish() because it would contravene (1).

Reply:
OK, this change is good. The only thing I'd say is that subclasses must not override publish() with a synchronized method AND ALSO call super.publish().